content/posts/cross-compile-query-json.mdx

```diff
@@ -0,0 +1,252 @@
+---
+title: "query-json: A story of cross-compilation with Reason"
+date: "2020-10-12"
+tags:
+  - "query-json"
+  - "json"
+  - "jq"
+  - "Reason"
+  - "OCaml"
+  - "js_of_ocaml"
+  - "jsoo"
+---
+
+<!-- TODO: Disclamer Difference between Rescript/Reason/BuckleScript/OCaml -->
+# What's query-json
+
+[query-json](https://github.com/davesnx/query-json) is a faster and simpler re-implementation of [jq's language](https://github.com/stedolan/jq/wiki/jq-Language-Description) in Reason Native.
+
+It's a CLI to run small programs against JSON, the same idea as [sed](https://www.linode.com/docs/tools-reference/tools/manipulate-text-from-the-command-line-with-sed/), but for JSON files. As a web engineer is an essential tool while developing APIs.
+
+Started with the goal to create something useful and learn during the process. Was very interested in learning about how to write parsers and compilers using the OCaml stack: [menhir](https://opam.ocaml.org/packages/menhir) and [sedlex](https://github.com/ocaml-community/sedlex), compile it to any architecture and try to compile it to JavaScript.
+
+This post explains the project, how it was made and which decisions where followed and some reflections.
+
+## Why learning parsers/compilers
+I had a little idea about the theory, but very vague and useless. Since I create [styled-ppx](https://github.com/davesnx/styled-ppx) which is a
```

Comment:
```suggestion
I had a little idea about the theory, but very vague and useless. Since I created [styled-ppx](https://github.com/davesnx/styled-ppx) which is a
```